src/main/fc/fc_core.c

```diff
@@ -350,11 +353,16 @@ void tryArm(void)
 #endif
     } else {
         if (!isFirstArmingGyroCalibrationRunning()) {
-            int armingDisabledReason = ffs(getArmingDisableFlags());
+            const armingDisableFlags_e armingDisableFlags = getArmingDisableFlags();
+            const int armingDisabledReason = ffs(armingDisableFlags);
             if (lastArmingDisabledReason != armingDisabledReason) {
                 lastArmingDisabledReason = armingDisabledReason;
-
-                beeperWarningBeeps(armingDisabledReason);
+                // if ARMING_DISABLED_RUNAWAY_TAKEOFF is set that should override other arming disabled beep patterns
```

Comment:
`armingDisableFlags` is ordered in order of priority. The desired result should be achieved by reordering the list to give `ARMING_DISABLED_RUNAWAY_TAKEOFF` the desired (high) priority. This will also avoid having to add this special case handling in two (and potentially more in the future) places.